### ISSUE TITLE
04_dimension_reduction_and_performance throws a multiprocessing exception on windows

### ISSUE THREAD

**Issue by amy12xx** [dirty-cat/dirty_cat#145]
This is due to the resource_used function used in this line:
results = resource_used(model_selection.cross_validate)(model, df, y, )

This can be fixed by having this inside the if __name__ == '__main__': block. I tested this on windows, and can confirm that doing this fixes the problem.

See here: https://stackoverflow.com/questions/18204782/runtimeerror-on-windows-trying-python-multiprocessing 

Details of exception raised:
File "C:\Users\Amanda\Miniconda3\envs\dirtycat\lib\multiprocessing\spawn.py", line 134, in _check_not_importing_main
    raise RuntimeError('''
RuntimeError: 
        An attempt has been made to start a new process before the
        current process has finished its bootstrapping phase.

        This probably means that you are not using fork to start your
        child processes and you have forgotten to use the proper idiom
        in the main module:

            if __name__ == '__main__':
                freeze_support()
                ...

        The "freeze_support()" line can be omitted if the program
        is not going to be frozen to produce an executable.

**Comment by GaelVaroquaux** [comment]
Sounds good! Let's add a " if __name__ == '__main__':".

@Phaide , do you want to look at this when you have time?

**Comment by Phaide** [comment]
Hi all,
Very good idea @GaelVaroquaux, I know this error very well :)
I'll have a look at the code on Wednesday !

**Comment by Phaide** [comment]
Hi all,
Sorry for the delay, I postponed this issue a little bit as I'm working on the fetching system these days.
Hopefully I'll make the PR next week and have a look at this right after :)
Thanks

**Comment by Phaide** [comment]
Hi,
By the way, I had a ``MemoryError`` when loading the dataset for the first time.
The traceback:
```
C:\Users\Phaide\AppData\Local\Programs\Python\Python38-32\lib\site-packages\dirty_cat\datasets\fetching.py:207: UserWarning: content size cannot be found, downloading file from https://data.montgomerycountymd.gov/api/views/4mse-ku6q/rows.csv?accessType=DOWNLOAD as a whole
  warnings.warn('content size cannot be found, '
Traceback (most recent call last):
  File "C:\Users\Phaide\Downloads\04_dimension_reduction_and_performance.py", line 50, in <module>
    data = fetch_traffic_violations()
  File "C:\Users\Phaide\AppData\Local\Programs\Python\Python38-32\lib\site-packages\dirty_cat\datasets\fetching.py", line 472, in fetch_traffic_violations
    return fetch_dataset(TRAFFIC_VIOLATIONS_CONFIG, show_progress=False)
  File "C:\Users\Phaide\AppData\Local\Programs\Python\Python38-32\lib\site-packages\dirty_cat\datasets\fetching.py", line 221, in fetch_dataset
    _fetch_file(urlinfo.url, data_dir, filenames=urlinfo.filenames,
  File "C:\Users\Phaide\AppData\Local\Programs\Python\Python38-32\lib\site-packages\dirty_cat\datasets\fetching.py", line 316, in _fetch_file
    _download_and_write(url, temp_full_name, show_progress=show_progress)
  File "C:\Users\Phaide\AppData\Local\Programs\Python\Python38-32\lib\site-packages\dirty_cat\datasets\fetching.py", line 211, in _download_and_write
    local_file.write(r.content)
  File "C:\Users\Phaide\AppData\Local\Programs\Python\Python38-32\lib\site-packages\requests\models.py", line 828, in content
    self._content = b''.join(self.iter_content(CONTENT_CHUNK_SIZE)) or b''
MemoryError
```
Solved by installing Python 3.8.6 x64.
